web/api/v1/api.go

```diff
@@ -174,6 +185,84 @@ func (api *API) labelValues(r *http.Request) (interface{}, *apiError) {
 	return vals, nil
 }
 
+var patLabelMatcher = regexp.MustCompile(`([a-zA-Z_][a-zA-Z0-9_]*)(=|!=|=~|!~)"(.*)"`)
```

Comment:
I think this needs to do a full string match rather than a substring match, so add `^...$`? Otherwise your form argument could have garbage after the closing quote at the end and it would still match.